### PR TITLE
Fixes #14: Adds _geographic_crs

### LINKED ISSUE #14: TMS JSON should specify the '_geographic_crs' field
To support different TMS in mosaic.json specification, it is necessary to explicitly provide the `_geographic_crs` field. By including, the rest of the tooling is able to instantiate the transformations between projected space and geographic space. 

[Here](https://github.com/jlaura/cogeo-mosaic/blob/mars_tms/tests/test_planetcantile.py) is a test using sueprmorecando with a planetary TMS. This will fail [here](https://github.com/developmentseed/morecantile/blob/main/morecantile/models.py#L162) when morecantile attempts to access the `_geographic_crs` field in the input data. This will default to None and then `proj` will throw the error that no transformation exists between our planetary CRS and WGS84. 

I think we can fix this in a number of places. The easiest integration is to provide that key on the planetcanetile side and use all of the other tooling as it already exists.

### PR DESCRIPTION
**Pull request by jlaura**
This PR simplifies the read/write on the TMS as JSON and adds support for encoding the `_geographic_crs` key. See #14 for a description of why this field is needed.

### CODE REVIEW

File: planetcantile/data/generate.py [hunk 1]
```diff
@@ -132,15 +132,14 @@ def __post_init__(self):
         pass
         #print(f'Could not find authority for {crs_obj.to_wkt()}')
 
-
-for tmsp in crss:
+for tmsp in crss[10:]:
```
Comment (AndrewAnnex) [inline planetcantile/data/generate.py]: why was this line changed

File: planetcantile/data/generate.py [hunk 2]
```diff
@@ -132,15 +132,14 @@ def __post_init__(self):
...
     # create the tms object
     tms = morecantile.TileMatrixSet.custom(**asdict(tmsp))
-    tmsj = tms.json(exclude_none=True)
-    tmsj = tmsj.replace(CRS_to_uri(tms.supportedCRS), CRS_to_urn(tms.supportedCRS), 1)
-    tmsj = tmsj.replace(CRS_to_uri(tms.boundingBox.crs), CRS_to_urn(tms.boundingBox.crs), 1)
-    _d = json.loads(tmsj)  # get to pretty printed json
+    tmsj = tms.dict()
```
Comment (AndrewAnnex) [inline planetcantile/data/generate.py]: I am not a fan of this change as it causes a huge diff by adding the "keywords", "title", "abstract", and "wellKnownScaleSet" json keys to output. See if it is possible to just set/override the `_geographical_crs` on the TMS object before the call to `.json`

File: planetcantile/data/generate.py [hunk 3]
```diff
@@ -132,15 +132,14 @@ def __post_init__(self):
...
+    # Include URN to the planetary projections; _geographic_crs is needed by downstream libs, e.g., morecantile 
+    tmsj['supportedCRS'] = tmsj['boundingBox']['crs'] = CRS_to_urn(tms.supportedCRS)
```
Comment (AndrewAnnex) [inline planetcantile/data/generate.py:140]: I may be missing something about dict updates, but this should probably through an exception of some kind due to the extra equals sign, or maybe it sets it to null? should be possible to remove by operating on the object directly as suggested above

**Review by AndrewAnnex — CHANGES_REQUESTED** [review]
see review comments, in particular concerned about the change to line 135 and 140

File: planetcantile/data/generate.py (hunk 1 above)
Comment (jlaura) [reply planetcantile/data/generate.py]: It shouldn't be. I was seeing errors in crss>10. I'll revert.

File: planetcantile/data/generate.py (hunk 2 above)
Comment (jlaura) [reply planetcantile/data/generate.py]: I tried operating directly on the object first and it was not possible. The older code was doing a straight string replacement, so I assumed you had run into the same issues with the object and solved it that way. This felt less hacky, though not through the object.

File: planetcantile/data/generate.py (hunk 3 above)
Comment (jlaura) [reply planetcantile/data/generate.py:140]: I thought (?) that this is pretty standard notation when setting two variables to the same value.

```python
>>> x = y = 2
>>> print(x)
2
>>> print(y)
2
```
Also, through the object was a no go.

Comment (AndrewAnnex) [reply planetcantile/data/generate.py:140]: huh interesting, I guess I never needed to set variables to the same value before like this.

Comment (AndrewAnnex) [inline planetcantile/data/generate.py:140]: okay so I am not sure if this correct behavior yet as I can see that the scales and boundingbox crss are different then how I had them defined in the diff for this pr. I think I decided to always specify the bounding boxes in the `naifid00` coordinates and this pr changes that behavior. Some tests should be added to verify that the correct scale set is being set and verify that the correct boundingbox crs is used

**Comment by AndrewAnnex** [comment]
@jlaura looking at this a bit harder, this may actually really be a bug with morecantile's parsing of the files and the init method for tilematrixset. Within the for-loop in planetcantile the `_geographic_crs` is correctly set. While updating the json files to include `_geographic_crs` does work, probably morecantile can/should be updated here https://github.com/developmentseed/morecantile/blob/644d3b70547420222d3c34071f2eb78da0e37646/morecantile/models.py#L162

File: planetcantile/data/generate.py (hunk 2 above)
Comment (AndrewAnnex) [inline planetcantile/data/generate.py]: replace with `tmsj = tms.dict(exclude_none=True)` to minify diff

**Comment by AndrewAnnex** [comment]
@jlaura mind if I push some fixes to the pr?

**Comment by jlaura** [comment]
@AndrewAnnex  Not at all! I'm wrapping a data release and was hoping to have time at the end of the week. Any attention you have for this is absolutely appreciated!
